react/src/lib/Popover/index.js

```diff
@@ -216,8 +274,41 @@ class Popover extends React.Component {
               },
               onMouseLeave: e => {
                 e.persist();
-                this.setState({ isHovering: false }, () => this.delayedHide(e));
+                this.handleHide(e);
               },
+              onKeyDown: (e) => {
```

Comment:
Since we don't care about any of this if it's not a shift/tab/esc, should we be doing any of these intensive processes if that's not the case?